OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -314,14 +328,31 @@ public HackyAI(HackyAIInfo info, ActorInitializer init)
 					&& !unit.Info.HasTraitInfo<HuskInfo>()
 					&& unit.Info.HasTraitInfo<ITargetableInfo>();
 
-			unitCannotBeOrdered = a => a.Owner != Player || a.IsDead || !a.IsInWorld;
+			unitCannotBeOrdered = UnitCannotBeOrdered;
 
 			foreach (var decision in info.PowerDecisions)
 				powerDecisions.Add(decision.OrderName, decision);
 
 			maximumCaptureTargetOptions = Math.Max(1, Info.MaximumCaptureTargetOptions);
 		}
 
+		bool UnitCannotBeOrdered(Actor a)
+		{
+			if (a.Owner != Player || a.IsDead || !a.IsInWorld)
+				return true;
+
+			var tag = a.TraitOrDefault<ScriptTags>();
```

Comment:
(carrying on the earlier discussion)

After spending a bit more time understanding your implementation:
* Using `ScriptTags` to read and write tags from AI scripts is good.  The AI is now using a different context to the map script so we don't need to worry about namespace conflicts.
* Using hardcoded tags set in lua and read by code is bad.  We already have a standard system for passing information between code and lua (the `*Properties` API bindings and triggers), so use it!